### ISSUE TITLE
switch non-autotools buildsystem for externs to pd-lib-builder

### ISSUE THREAD

**Issue by umlaeute** [pure-data/pure-data#2284]
i noticed that the github actions didn't cause the MinGW build failure recently introduced (and already fixed with 18327971c596c18b2d41f0ea8d9bbca55f44cde9).
while investigating why, i noticed that the default `all` target of `src/makefile.mingw` does not build externals https://github.com/pure-data/pure-data/blob/96417889c8a339fcb262e2c632b64540489ec16b/src/makefile.mingw#L171

unlike the default `all` targets of the macOS resp GNU makefiles, e.g. https://github.com/pure-data/pure-data/blob/d9136987d6664cb33d9e22249ccb9fafe6ab8198/src/makefile.gnu#L156

on closer inspection, this is probably because the `externs` target of the MinGW makefile is rather simplistic https://github.com/pure-data/pure-data/blob/81f1eecd42f65aa8a3ab3c246550313a3a6c7542/src/makefile.mingw#L231-L232 unlike the more explicit targets found in the other variants https://github.com/pure-data/pure-data/blob/3810461b9faf5c0d2559bcd0bb6e059543e5aa71/src/makefile.gnu#L184-L194

needless to say, that there is no `all` target in `../externs/` (if not using autotools, there isn't even a makefile)

also, the Windows part of the common makefile snippet is really targetted at MSVC https://github.com/pure-data/pure-data/blob/18327971c596c18b2d41f0ea8d9bbca55f44cde9/extra/makefile.subdir#L9-L33 and does not work with MinGW.

rather than trying to fix this, i would suggest switching the pure-make (non-autotools) buildsystems of the externals in `externs/` to the tried and tested `pd-lib-builder` (which would also act as a documentation on how to properly build externals).l

the same goes for `doc/6.externs/`

**Comment by danomatika** [comment]
I agree. Maintaining the various build systems should be easier than it is. It's hard sometimes to remember how many places need to be updated even though they generally all end up using the same per-platform flags...
